master/buildbot/changes/gitpoller.py

```diff
@@ -311,6 +312,16 @@ def _convert_nonzero_to_failure(res,
             "utility to handle the result of getProcessOutputAndValue"
             (stdout, stderr, code) = res
             if code != 0:
+                if code == 128:
+                    # 128 is the error code git returns when things are
+                    # wrong on its end
+                    error_message = ('RuntimeError: command %s %s in %s on '
+                                    'repourl %s failed with exit code %d: %s'
+                                    % (command, args, path, self.repourl,
+                                       code, stderr)).strip()
+                    sys.stderr.write(error_message)
+                    sys.stderr.flush()
+                    sys.exit(1)
```

Comment:
I am not sure about `sys.exit(1)` here -- this means that should something get broken while master is running, it will suddenly stop.